src/components/2FA/index.tsx

```diff
@@ -20,43 +21,41 @@ import {
 } from './styles'
 
 type Props = {
-  qrcode: string
   userId: string
   handleCloseModal: VoidFunction
   onSuccessCallback: FunctionWithParams<AuthResponse>
   onErrorCallback?: FunctionWithParams<string>
 }
 
 export const User2FAComponent: FC<Props> = ({
-  qrcode,
   userId,
   onSuccessCallback,
   handleCloseModal,
   onErrorCallback = noopFunction
 }) => {
-  const [attemptCount, setAttempCount] = useState(3)
-  const [code, setCode] = useState('')
+  const [attemptLeftCount, setAttemptLeftCount] = useState(3)
+  const [secure2FACode, setSecure2FACode] = useState('')
   const [submit2faData, { isLoading, isError, isSuccess, data }] = useConfirmUser2FAMutation()
 
   useEffect(() => {
-    if (attemptCount <= 0) {
+    if (attemptLeftCount <= 0) {
       handleCloseModal()
-      onErrorCallback(AUTH_FAILURE_MESSAGE)
+      onErrorCallback(ErrorMessages.FAILURE_AUTH_ERROR)
     }
-  }, [attemptCount, onErrorCallback, handleCloseModal])
+  }, [attemptLeftCount, onErrorCallback, handleCloseModal])
 
   useEffect(() => {
     if (isError) {
-      setAttempCount((prev) => --prev)
+      setAttemptLeftCount((prev) => --prev)
```

Comment:
should this be controlled by the server?

Reply:
if user will reload the page - error count will be refreshed

Reply:
Was added this ability on server and client